### DISCUSSION TITLE
Add accessibility for the visually impaired

### DISCUSSION THREAD

**Discussion by Mudb0y** [ppy/osu#23925] [Ideas]
I recently found-out about Osu, and decided to mess with Laser today. The game looks really cool!
I'm blind, and it'd be cool to have another rhythm game to play, especially because of all the content from some of my favourite artists that is available here.
Since Osu is now being rewritten, I was wondering if it'd be with-in the realm of possibility to implement at least some degree of accessibility in to the game?
At the basic level, at least making better keyboardd navigability is required. I downloaded the game today, and could read the screen with OCR, however I couldn't really figure-out how to choose options in the menus with the keyboard at all. This shouldn't be hard to implement, if you choose to do so.
Another important part would be built-in screen-reading. Depending on what you use for your UI the degree of difficulty with implementing speech / screen-reading support varies, but at best it's going to be quite a bit of work I imagine.
Now, I've never played Osu my self, and still need to read-up more on how the actual gameplay mechanics work, but most rhythm games I've ever played, including Nintendo's Rhythm Tengoku / Rhythm Heaven series and an indie game called Sequence Storm use audio queues to represent what you have to do. For example, when you have to press the spacebar, a certain sound plays ahead of time to let you know the action is incoming.
This is just an idea, and as the game is deep with-in development I'm not really expecting much to be possible, given that you guys have higher priorities I'm sure, but I'd love it if you could at least try doing something in this regard.
If you need any clarification, help, ETC do let me know and I'll do my best!

**Comment by peppy** [comment]
Our framework is written from scratch. I highly doubt we will be able to add full accessibility support, sorry.

As for navigating the menus, most are already keyboard navigable using things like space bar, enter, escape, arrow keys and shortcuts like ctrl 1-4 (to change rulesets). You can already get most of the way through, with the exception of loading beatmaps into the game (but you might be able to figure that out from your browser):

What you'll most have trouble with is gameplay, as it basically requires vision in all rulesets we have implemented.
